### PR TITLE
[GPU] Swiglu OPT kernel implementation

### PR DESCRIPTION
**Pull request by dnkurek**
### Details:
 - Implement Swiglu OPT kernel
 - Use fast erf function instead of slow OpenCL erf

### Tickets:
 - 149484

### CODE REVIEW

**Comment by e-ddykim** [comment]
Is there a reason to add a new opt kernel without applying `fast_erf` to the ref kernel?

**Comment by dnkurek** [comment]
> Is there a reason to add a new opt kernel without applying `fast_erf` to the ref kernel?

Well, while fast_erf is the most important part of this performance-wise, this opt kernel has a simpler layout and later on it can be easily modified to support block reads (which I did and it did give more performance boost aswell, but it's more work which can be done later)

So it's just convenience in my opinion, plus it's slightly faster.

**Comment by dnkurek** [comment]
> Is there a reason to add a new opt kernel without applying `fast_erf` to the ref kernel?

Plus, also a faster implementation for tanh will be needed in the future, but it's out of scope of this PR

**Comment by dnkurek** [comment]
> Is there a reason to add a new opt kernel without applying `fast_erf` to the ref kernel?

Also, since this kernel has grown to support more than just swiglu, wouldn't you think it would make sense to rename this operation to something like "gated activation" in a later PR? I think the name now is a bit misleading.

**Comment by e-ddykim** [comment]
> > Is there a reason to add a new opt kernel without applying `fast_erf` to the ref kernel?
> 
> Also, since this kernel has grown to support more than just swiglu, wouldn't you think it would make sense to rename this operation to something like "gated activation" in a later PR? I think the name now is a bit misleading.

Yes, I agree with you. "gated activation" or "GLU activation" would be better than the current name.

File: src/plugins/intel_gpu/src/kernel_selector/cl_kernels/swiglu_gpu_opt.cl [hunk 1]
```diff
@@ -0,0 +1,51 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+// From A&S Handbook of Mathematical Functions with Formulas, Graphs, and Mathematical Tables
+// Formula section 7.1.26
+
+ACCUMULATOR_TYPE fast_erf(ACCUMULATOR_TYPE x) {
```
Comment (sshlyapn) [inline src/plugins/intel_gpu/src/kernel_selector/cl_kernels/swiglu_gpu_opt.cl:8]: Please use wrappers for functions inside kernels: `FUNC(fast_erf)(ACCUMULATOR_TYPE x)`
With related `FUNC_CALL(fast_erf)(x)` call

File: src/plugins/intel_gpu/src/kernel_selector/cl_kernels/swiglu_gpu_opt.cl [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    const ACCUMULATOR_TYPE a1 = 0.254829592f;
```
Comment (sshlyapn) [inline src/plugins/intel_gpu/src/kernel_selector/cl_kernels/swiglu_gpu_opt.cl:9]: Should these coefficients be of `ACCUMULATOR_TYPE`? Integer types can be used as accumulators too (`GetAccumulatorType()` function allows them)

Comment (dnkurek) [reply src/plugins/intel_gpu/src/kernel_selector/cl_kernels/swiglu_gpu_opt.cl:9]: We are dealing with ACCUMULATOR_TYPE inputs so I guess it also makes sense that the coefficients would be of ACCUMULATOR_TYPE, unless I am missing something

File: src/plugins/intel_gpu/src/kernel_selector/cl_kernels/swiglu_gpu_opt.cl (hunk 1 above)
Comment (dnkurek) [reply src/plugins/intel_gpu/src/kernel_selector/cl_kernels/swiglu_gpu_opt.cl:8]: Acknowledged

**Comment by dnkurek** [comment]
Current fast_erf function does have bad accuracy when x is close to 0, I am currently researching better formulas which are more appropiate for this use case

**Review by Lyamin-Roman — APPROVED** [review]

**Review by sshlyapn — APPROVED** [review]

**Pull request merged**